src/coap_openssl.c

```diff
@@ -638,19 +1242,47 @@ int coap_dtls_context_set_pki( coap_context_t *ctx,
     SSL_set_options(context->dtls.ssl, SSL_OP_COOKIE_EXCHANGE);
     SSL_set_mtu(context->dtls.ssl, COAP_DEFAULT_MTU);
   }
-  context->psk_pki_enabled = 1;
+  context->psk_pki_enabled |= IS_PKI;
   return 1;
 }
 
-int coap_dtls_context_check_keys_enabled(coap_context_t *ctx)
+int
+coap_dtls_context_set_pki_root_cas(struct coap_context_t *ctx,
+                                   const char *ca_file,
+                                   const char *ca_path
+) {
+  coap_openssl_context_t *context =
+                                ((coap_openssl_context_t *)ctx->dtls_context);
+  if (context->dtls.ctx) {
+    if (!SSL_CTX_load_verify_locations(context->dtls.ctx, ca_file, ca_path)) {
+      coap_log(LOG_WARNING, "Unable to install root CAs (%s/%s)\n",
+               ca_file ? ca_file : "NULL", ca_path ? ca_path : "NULL");
+      return 0;
+    }
+  }
+  if (context->tls.ctx) {
+    if (!SSL_CTX_load_verify_locations(context->tls.ctx, ca_file, ca_path)) {
+      coap_log(LOG_WARNING, "Unable to install root CAs (%s/%s)\n",
+               ca_file ? ca_file : "NULL", ca_path ? ca_path : "NULL");
+      return 0;
+    }
+  }
+  return 1;
+}
+
+int
+coap_dtls_context_check_keys_enabled(coap_context_t *ctx)
 {
-  coap_openssl_context_t *context = ((coap_openssl_context_t *)ctx->dtls_context);
+  coap_openssl_context_t *context =
+                                 ((coap_openssl_context_t *)ctx->dtls_context);
   return context->psk_pki_enabled;
 }
 
 
 void coap_dtls_free_context(void *handle) {
+  int i;
```

Comment:
Should be of type `size_t` or so in alignment with `context->sni_count.

Reply:
Fixed